mne_icalabel/megnet/features.py

```diff
@@ -0,0 +1,169 @@
+import io
+
+import matplotlib.pyplot as plt
+import mne  # type: ignore
+import numpy as np
+from mne.io import BaseRaw  # type: ignore
+from mne.preprocessing import ICA  # type: ignore
+from mne.utils import warn  # type: ignore
+from numpy.typing import NDArray
+from PIL import Image
+from scipy import interpolate  # type: ignore
+from scipy.spatial import ConvexHull  # type: ignore
+
+from ._utils import cart2sph, pol2cart
+
+
+def get_megnet_features(raw: BaseRaw, ica: ICA):
+    """Extract time series and topomaps for each ICA component.
+
+    the main work is focused on making BrainStorm-like topomaps
+    which trained the MEGnet.
+
+    Parameters
+    ----------
+    raw : BaseRaw
+        The raw MEG data. The raw instance should have 250 Hz
+        sampling frequency and more than 60 seconds.
+    ica : ICA
+        The ICA object containing the independent components.
+
+    Returns
+    -------
+    time_series : np.ndarray
+        The time series for each ICA component.
+    topomaps : np.ndarray
+        The topomaps for each ICA component
+
+    """
+    if "meg" not in raw:
+        raise RuntimeError(
+            "Could not find MEG channels in the provided "
+            "Raw instance. The MEGnet model was fitted on"
+            "MEG data and is not suited for other types of channels."
+        )
+
+    if raw.times[-1] < 60:
+        raise RuntimeError(
+            f"The provided raw instance has {raw.times[-1]} seconds. "
+            "MEGnet was designed to classify features extracted from "
+            "an MEG datasetat least 60 seconds long. "
+        )
+
+    if not np.isclose(raw.info["sfreq"], 250, atol=1e-1):
```

Comment:
Would it be better to use warn instead of raise in the presence of linenoise? Based on my experience, linenoise does not affect ICA's ability to handle eye movements, saccades, and heartbeat ICs, so it should not impact MEGnet's performance either, as MEGnet is only sensitive to these artifacts.